### PR TITLE
Add `expectSigned` option to `openpgp.decrypt` and `openpgp.verify`

### LINKED ISSUE #1274: Add `expectSigned` option to enforce signature verification in openpgp.decrypt
`openpgp.decrypt` does not throw if signatures are missing or do not verify. Instead, one needs to inspect the returned `{ signatures }` object for errors/validity. This behaviour makes sense as a message is not necessarily signed, or the recipient might not have the (correct) public keys to verify the signature.
However, it can result in verification errors being missed. To help avoid such issues, as well as simplify dealing with messages that are expected to be signed, we could add an `expectSigned : Boolean` option that - if set - results in `decrypt` throwing an error and not returning any data if the message signatures are missing or invalid.

### PR DESCRIPTION
**Pull request by larabr**
Fix #1274 , changes: 
- add `expectSigned` option to `openpgp,decrypt` and `openpgp.verify` (defaulting to false)
- if `expectSigned = true`:
  - `openpgp.decrypt` throws immediately if public keys or signatures are missing, or if the decrypted data is not streamed & signatures are invalid
  - `openpgp.verify` throws immediately if signatures are missing, or if the decrypted data is not streamed & signatures are invalid
  - if the data is streamed, and signatures are invalid, reading the data stream will eventually throw

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -267,6 +268,41 @@ Encrypt with multiple public keys:
...
+
+    const message = await openpgp.readMessage({
+        armoredMessage: encryptedAndSignedMessage // parse armored message
+    });
+    // decryption will fail if all signatures are invalid or missing
+    const { data: decrypted, signatures } = await openpgp.decrypt({
+        message,
+        privateKeys: privateKey // for decryption
+        expectSign: true,
+        publicKeys: publicKey, // for verification (mandatory with expectSign=true)
```
Comment (twiss) [inline README.md:300]:
```suggestion
        expectSigned: true,
        publicKeys: publicKey, // for verification (mandatory with expectSigned=true)
```

File: src/openpgp.js [hunk 2]
```diff
@@ -311,6 +312,21 @@ export function decrypt({ message, privateKeys, passwords, sessionKeys, publicKe
     result.data = format === 'binary' ? decrypted.getLiteralData() : decrypted.getText();
     result.filename = decrypted.getFilename();
     linkStreams(result, message);
+    if (expectSigned) {
+      if (publicKeys.length === 0) {
+        throw new Error('public keys are required to verify message signatures');
```
Comment (twiss) [inline src/openpgp.js:317]: This is very nitpicky but please capitalize these error messages
```suggestion
        throw new Error('Public keys are required to verify message signatures');
```
(also the other ones)

File: src/openpgp.js [hunk 3]
```diff
@@ -311,6 +312,21 @@ export function decrypt({ message, privateKeys, passwords, sessionKeys, publicKe
...
+      }
+      if (result.signatures.length === 0) {
+        throw new Error('message is not signed');
+      }
+      result.data = stream.concat([
+        result.data,
+        stream.fromAsync(() => util.anyPromise(
+          result.signatures.map(sig => sig.verified)
+        ).then(() => {}) // append nothing to the data
+          .catch(() => { throw new Error('invalid message signature'); }))
```
Comment (twiss) [inline src/openpgp.js:327]: I know this was my suggestion but probably it's cleaner to write it like
```suggestion
        stream.fromAsync(async () => {
          await util.anyPromise(result.signatures.map(sig => sig.verified));
        })
```

File: src/openpgp.js (hunk 2 above)
Comment (larabr) [reply src/openpgp.js:317]: But these errors are wrapped and appear after "Error decrypting message:"

Comment (twiss) [reply src/openpgp.js:317]: Yeah, but the same thing is true for most other errors in this library, and we capitalize those

**Review by twiss — APPROVED** [review]

**Pull request merged**